src/core/mac/sub_mac.cpp

```diff
@@ -1004,6 +1034,28 @@ void SubMac::HandleCslTimer(void)
         break;
     }
 }
+
+void SubMac::GetCslWindowEdges(uint32_t *ahead, uint32_t *after)
+{
+    uint32_t semiPeriod = mCslPeriod * kUsPerTenSymbols / 2;
+    uint64_t curTime    = otPlatRadioGetNow(&GetInstance());
+    uint32_t elapsed, semiWindow;
+
+    if (mCslLastSync.GetValue() > curTime)
+    {
+        elapsed = static_cast<uint32_t>(UINT64_MAX - mCslLastSync.GetValue() + curTime);
+    }
+    else
+    {
+        elapsed = static_cast<uint32_t>(curTime - mCslLastSync.GetValue());
+    }
+
+    semiWindow = elapsed * (Get<Radio>().GetCslAccuracy() + mCslParentDrift) / 1000000;
+    semiWindow += mCslParentUncert * 100;
```

Comment:
Why is `mCslParentUncert`'s unit 100us? Can we add a const enum for it? For example, `kUsPerUncertUnit`.

Reply:
Currently 100 us is agreed, but I'm proposing to use units of 10 us: https://threadgroup.atlassian.net/browse/SPEC-961?focusedCommentId=49637

Reply:
BTW, it would be a good exercise to identify the required uncertainty value in OT when transmit_at is not used.

Reply:
> it would be a good exercise to identify the required uncertainty value in OT

Uncertainty depends on application. You cannot identify constant for a network stack. The easiest way would be to add public API to OT so that application enabling Thread 1.2 router would provide this value. Alternatively, OT could try to measure uncertainty, and select probable "worst case" using statistical analysis.